quetz/pkgstores.py

```diff
@@ -376,7 +380,11 @@ def remove_prefix(text, prefix):
         with self._get_fs() as fs:
             return [remove_prefix(f, channel_bucket) for f in fs.find(channel_bucket)]
 
-    def url(self, channel: str, src: str, expires=3600):
+    def url(
+        self, channel: str, src: str, expires=3600, content_type: Optional[str] = None
+    ):
+        if content_type is not None:
+            raise NotImplementedError("'content_type' is not supported for S3")
```

Comment:
TODO: Check what to do with S3. I am not sure if S3 will happily accept non-signed headers or will error out (in the latter case we can't use redirects for `repodata.json` on S3)